panic: runtime error: invalid memory address or nil pointer dereference in variables.go

OS: Mac 10.12.12
Delve: 0.12.1
Go: 1.7.4 darwin/amd64

Running Go in VSCode. All that happens, upon start, is I get this stacktrace. I have some args in my launch json (at the bottom) but this happens with or without.


Stacktrace:
```
2017/02/10 22:28:51 server.go:73: Using API v1
2017/02/10 22:28:51 debugger.go:68: launching process with args: [/Users/nathanhyland/gocode/src/github.com/natdm/goflow/debug -file ./testdata/fixtures.go -out ./testdata/]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x4220f75]
goroutine 1 [running]:
panic
(0x44b4f40, 0xc4200140b0)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/derekparker/delve/pkg/proc.(*EvalScope).DwarfReader(0xc4208d2d20, 0xc420055710)
	/Users/nathanhyland/gocode/src/github.com/derekparker/delve/pkg/proc/variables.go:324 +0x25
github.com/derekparker/delve/pkg/proc.(*EvalScope).packageVarAddr(0xc4208d2d20, 0x454220c, 0x14, 0x0, 0x0, 0x44af220)
	/Users/nathanhyland/gocode/src/github.com/derekparker/delve/pkg/proc/variables.go:663 +0x35
github.com/derekparker/delve/pkg/proc.(*Process).EvalPackageVariable(0xc420174140, 0x454220c, 0x14, 0x1, 0x0, 0x40, 0x0, 0x0, 0xc420150600, 0xc42015a6c0, ...)
	/Users/nathanhyland/gocode/src/github.com/derekparker/delve/pkg/proc/variables.go:654 +0x84
github.com/derekparker/delve/pkg/proc.(*Process).getGoInformation(0xc420174140, 0x0, 0x0
, 0x0, 0x0, 0x0, 0x0, 0x0, 0x2403044af220)
	/Users/nathanhyland/gocode/src/github.com/derekparker/delve/pkg/proc/proc.go:894 +0xbc
github.com/derekparker/delve/pkg/proc.initializeDebugProcess(0xc420174140, 0xc42014e6c0, 0x3c, 0x0, 0x0, 0x0, 0xc42015e230)
	/Users/nathanhyland/gocode/src/github.com/derekparker/delve/pkg/proc/proc.go:834 +0xf4
github.com/derekparker/delve/pkg/proc.Launch(0xc420150540, 
0x3, 0x3, 0x4535075, 0x1, 0x48044b0, 0x0, 0x49)
	/Users/nathanhyland/gocode/src/github.com/derekparker/delve/pkg/proc/proc_darwin.go:121 +0x86e
github.com/derekparker/delve/service/debugger.New(0xc4201505d0, 0xc4201505d0, 0xc420055a10, 0x40117e8)
	/Users/nathanhyland/gocode/src/github.com/derekparker/delve/service/debugger/debugger.go:69
 +0x2b7
github.com/derekparker/delve/service/rpccommon.(*ServerImpl).Run(0xc42015e320, 0xc42015e201, 0xc42015e320)
	/Users/nathanhyland/gocode/src/github.com/derekparker/delve/service/rpccommon/server.go:117 +0x13d
github.com/derekparker/delve/cmd/dlv/cmds.execute(0x0, 0xc420150540, 0x3, 0x3, 0xc42014c120
, 0x1, 0x0)
	/Users/nathanhyland/gocode/src/github.com/derekparker/delve/cmd/dlv/cmds/commands.go:416 +0x1e6
github.com/derekparker/delve/cmd/dlv/cmds.debugCmd.func1(0xc420160a00, 0xc4201585a0, 0x2, 0x6, 0x0)
	/Users/nathanhyland/gocode/src/github.com/derekparker/delve/cmd/dlv/cmds/commands.go:243 +0x4c3
github.com/derekparker/delve/cmd/dlv/cmds.debugCmd(0xc420160a00, 0xc4201585a0, 0x2, 
0x6)
	/Users/nathanhyland/gocode/src/github.com/derekparker/delve/cmd/dlv/cmds/commands.go:244 +0x49
github.com/derekparker/delve/vendor/github.com/spf13/cobra.(*Command).execute(0xc420160a00, 0xc4201584e0, 0x6, 0x6, 0xc420160a00, 0xc4201584e0)
	/Users/nathanhyland/gocode/src/github.com/derekparker/delve/vendor/github.com/spf13/cobra/command.go:569 +0x411
github.com/derekparker/delve/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420160400, 0xc420160a00, 0xc420160800, 0xc420160600)
	/Users/nathanhyland/gocode/src/github.com/derekparker/delve/vendor/github.com/spf13/cobra/command.go:656 +0x367
github.com/derekparker/delve/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420160400, 0xc420055f90, 0xc4200001a0)
	/Users/nathanhyland/gocode/src/github.com/derekparker/delve/vendor/github.com/spf13/cobra/command.go:615
 +0x2b
main.main()
	/Users/nathanhyland/gocode/src/github.com/derekparker/delve/cmd/dlv/main.go:13 +0x55
```

Launch.json:
```json
{
	"version": "0.2.0",
	"configurations": [
		{
			"name": "Golang",
			"type": "go",
			"request": "launch",
			"mode": "debug",
			"remotePath": "",
			"port": 2345,
			"host": "127.0.0.1",
			"program": "${workspaceRoot}",
			"env": {},
			"args": ["-file ./testdata/fixtures.go", "-out ./testdata/"],
			"showLog": true
		}
	]
}
```

Even i am facing the same issue in debugging 

2017/02/13 15:44:57 server.go:73: Using API v1
2017/02/13 15:44:57 debugger.go:68: launching process with args: [/Users/aputta/GO_CODE/src/github.com/workflow-orchestrator/debug]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1
 addr=0x30 pc=0x4221795]
goroutine 1 [running]:
panic(0x44b5920, 0xc4200160a0)
	/usr/local/Cellar/go/1.7.5/libexec/src/runtime/panic.go:500 +0x1a1
github.com/derekparker/delve/pkg/proc.(*EvalScope).DwarfReader(0xc422766f80, 0xc4200ba800)
	/Users/aputta/GO_CODE/src/github.com/derekparker/delve/pkg/proc/variables.go:324 +0x25
github.com/derekparker/delve/pkg/proc.(*EvalScope).packageVarAddr(0xc422766f80, 0x4542bea, 0x14, 0x0, 0x0, 0x44afc00)
	/Users/aputta/GO_CODE/src/github.com/derekparker/delve/pkg/proc/variables.go:663 +0x35

My Launch.json is 

{
    "version": "0.2.0",
    "configurations": [
        
        {
            "name": "Launch",
            "type": "go",
            "request": "launch",
            "mode": "debug",
            "remotePath": "",
            "port": 2345,
            "host": "127.0.0.1",
            "program": "${workspaceRoot}",
            "env": {},
            "args": [],
            "showLog": true
        }
    ]
}
Please let us know on the release when the fixes would be available.

--- a/Makefile
+++ b/Makefile
@@ -64,7 +64,7 @@ else
 endif
 
 test-proc-run:
-	go test $(TEST_FLAGS) $(BUILD_FLAGS) -test.v -test.run="$(RUN)" $(PREFIX)/proc
+	go test $(TEST_FLAGS) $(BUILD_FLAGS) -test.v -test.run="$(RUN)" $(PREFIX)/pkg/proc
 
 test-integration-run:
-	go test $(TEST_FLAGS) $(BUILD_FLAGS) -test.run="$(RUN)" $(PREFIX)/service/test
+	go test $(TEST_FLAGS) $(BUILD_FLAGS) -test.run="$(RUN)" $(PREFIX)/pkg/service/test

--- a/pkg/proc/proc_darwin.go
+++ b/pkg/proc/proc_darwin.go
@@ -253,7 +253,7 @@ func (dbp *Process) addThread(port int, attach bool) (*Thread, error) {
 	}
 	dbp.threads[port] = thread
 	thread.os.threadAct = C.thread_act_t(port)
-	if dbp.CurrentThread == nil {
+	if dbp.currentThread == nil {
 		dbp.SwitchThread(thread.ID)
 	}
 	return thread, nil

